File: leetcode.com/1191_k_concatenation_maximum_sum/solution.py
class Solution:
    def kConcatenationMaxSum(self, arr: List[int], k: int) -> int:
        local_max = 0
        global_max = 0
        
        for n in arr * k:
            local_max = max(n, n + local_max)
            
            if local_max > global_max:
                global_max = local_max
                
        return global_max
 

File: leetcode.com/1191_k_concatenation_maximum_sum/solution.c
#include <stdio.h>

#define MAX(a, b) (a > b) ? a : b

int kConcatenationMaxSum(int* arr, int arrSize, int k)
{
    long long lmax = 0;
    long long gmax = 0;

    for (int i = 0 ; i < k ; i++) {
        for (int j = 0 ; j < arrSize ; j++) {
            lmax = MAX(arr[j], arr[j] + lmax);

            if (lmax > gmax)
                gmax = lmax;
        }
    }

    return gmax % 1000000007;
}

int main()
{
    int arr[] = {10000,10000,10000,10000,10000,10000,10000,10000,10000,10000};
    int k = 100000;

    printf("%d\n", kConcatenationMaxSum(arr, sizeof(arr) / sizeof(int), k));
}
